spec/services/quarterly_external_income_export_spec.rb

```diff
@@ -0,0 +1,47 @@
+RSpec.describe QuarterlyExternalIncomeExport do
+  let!(:delivery_partner) { create(:delivery_partner_organisation) }
+  let!(:fund) { create(:fund_activity, :newton) }
+  let(:export) { QuarterlyExternalIncomeExport.new(delivery_partner, fund.source_fund_code) }
+
+  let :quarter_headers do
+    export.headers.drop(7)
+  end
+
+  let :external_income_data do
+    export.rows.map { |row| row.take(1) + row.drop(7) }
+  end
+
+  it "exports one quarter of external income for a single project" do
+    project = create(:project_activity, :newton_funded, organisation: delivery_partner)
+
+    ei1 = create(:external_income, activity: project, financial_year: 2014, financial_quarter: 1, amount: 10)
+    create(:external_income, activity: project, financial_year: 2014, financial_quarter: 1, amount: 20)
```

Comment:
In the spirit of keeping our unit tests quick, can we use mocks and stubs rather than creating real objects in the database?

Reply:
Something like:

```
project = build(:project_activity, :newton_funded, organisation: delivery_partner)

relation = double("ActiveRecord::Relation")
allow(ExternalIncome).to receive(:activity, :organisation).and_return(relation)
allow(relation).to receive(:where).with(activity_id: [project.id]).and_return(relation)
allow(relation).to receive(:order).with(:activity_id, :financial_year, :financial_quarter).and_return([
  build(:external_income, activity: project, financial_year: 2014, financial_quarter: 1, amount: 10)
  build(:external_income, activity: project, financial_year: 2014, financial_quarter: 1, amount: 20)
])
```